UI/GardenDeskForm.cs

```diff
@@ -101,27 +92,293 @@ private async void BtnLogin_Click(object sender, EventArgs e)
+            // confirm that the employee's info has been updated
+            MessageBox.Show($"{selectedEmployee.FirstName} {selectedEmployee.LastName}'s information has been updated.");
+            ShowUsersView();
+        }
+        // prefill the textboxes and combobox
+        private void PrefillTextbox()
+        {
+            textBoxFirstName.Text = selectedEmployee.FirstName;
+            textBoxLastName.Text = selectedEmployee.LastName;
+            comboBoxTypeUser.SelectedItem = selectedEmployee.Role.ToString();
+            textBoxEmailAddress.Text = selectedEmployee.Email;
+            textBoxPhoneNumber.Text = selectedEmployee.PhoneNumber;
+            textBoxBranch.Text = selectedEmployee.Branch;
+        }
+        #endregion
+        #region Delete employee logic
+        // method to delete the employee (my poor amigo)
+        private async void btnDeleteEmployee_Click(object sender, EventArgs e)
+        {
+            if (ConfirmDelete())
+            {
+                await employeeService.DeleteEmployeeByID(selectedEmployee.EmployeeId);
+                // confirm that the employee was deleted
+                MessageBox.Show($"{selectedEmployee.FirstName} {selectedEmployee.LastName} will be missed...");
+                ShowUsersView();
+            }
+        }
+        // ask the user to confirm deleting the employee
+        private bool ConfirmDelete()
+        {
+            // add a message box to get confirmation for deleting them
+            string messageTop = "Confirmation";
+            string messageText = "Are you sure you wish to delete this employee?";
+            return MessageBox.Show(messageText, messageTop, MessageBoxButtons.YesNo) == DialogResult.Yes;
+        }
+        #endregion
+        private void btnCancelChangesEmployee_Click(object sender, EventArgs e)
+        {
+            ShowUsersView();
+        }
+        #region Users view and buttons control
+        // method to show the panel, refresh the employees list and the buttons
+        private async void ShowUsersView()
         {
             ShowPanel(pnlUsers);
+            ChangeButtonState(btnEditEmployee, Color.LightGray, false);
+            ChangeButtonState(btnAddEmployee, Color.LightGreen, true);
             await DisplayEmployeesAsync();
         }
 
-        private async void menuItemIncedents_Click(object sender, EventArgs e)
+        // method to make the edit employee button enabled once the user has selected an employee
+        // and make the add user btn disabled
+        private void usersList_SelectedIndexChanged(object sender, EventArgs e)
         {
-            ShowPanel(pnlTicketsOverview);
-            await DisplayTicketsAsync(currentEmployee);
+            if (usersList.SelectedItems.Count > 0)
+            {
+                ChangeButtonState(btnEditEmployee, Color.Orange, true);
+                ChangeButtonState(btnAddEmployee, Color.LightGray, false);
```

Comment:
You can deselect an employee. Clicking inside the users view but not on a filled row will deselect an employee. I can send a video showing that. I had decided that it makes more sense for the add button to be disabled, it doesn't make sense to add an employee while selecting another employee. I can think of a few options if you think Dan might find it confusing how deselecting a row works:
- I can either put a button that says "deselect" to deselect everything
- I can put a tool-tip over disabled buttons that shows how to enable it, which would be the way to deselect for an add button (I think this could be the most useful for all buttons and makes our app better quality. It would be a simple string parameter)
- Completely ignore this and make adding a new employee possible even when there's already an employee selected
Let me know which one you think would be better for the app.